src/Utilities/useGetEnvironment.ts

```diff
@@ -47,4 +52,15 @@ const onPremFlag = (flag: string): boolean => {
 
 // Since some of these flags are only relevant to
 // the service, we need a way of bypassing this for on-prem
-export const useFlag = !process.env.IS_ON_PREMISE ? useUnleashFlag : onPremFlag;
+export const useFlag = (flag: string) => {
+  const { isFedoraEnv } = useGetEnvironment();
+
+  const onPremFlagValue = onPremFlag(flag);
+  const unleashFlagValue = useUnleashFlag(flag);
```

Comment:
Should this be skipped for fedora and on-prem? I'm assuming this will fail in those environments, or it wont? :)

Reply:
We can't call a hook conditionally, but no worries, it just returns undefined in those env.

Reply:
Ah right! I always forget :)